### ISSUE TITLE
Support for array casting

### ISSUE THREAD

**Issue by fhusquinet** [64robots/nova-json#2]
Is there any way your package could add support for array casting?

I currently use a JSON column in my database which is cast to an array using the $casts attribute on my model. 

When using nova-json the value sent to the field is already a valid JSON string, which throws an error when it is being parsed. This is pretty easy to fix by simply checking if the field value is an object or a string before parsing it.

The second issue is a bit harder to change, which is why I did not send a PR.
When submitting the form the value is stringified to a json string which is then parsed again by Laravel (because of the mutator) before updating the database.

Would it be possible to add an option to the field saying which cast is used, if any?

If the field is casted to an array then the Vue component shouldn't be dealing with JSON encoding/decoding.

I currently use a custom getter/setter which encode/decode the field accordingly but it seems a bit overkill...

**Comment by beliolfa** [comment]
Sounds like a great enhancement. I'll take a look later and will try to implement array cast.

Thanks!

**Comment by shirshak55** [comment]
Yea without casting this is creating too much problem :(

**Comment by beliolfa** [comment]
This has been fixed in 1.0.4.

Regarding to saving decoded data you could use this:

```php
->fillUsing(function($request, $model, $attribute, $requestAttribute) {
                     $decoded_payload = json_decode($request[$requestAttribute]);
                     // your logic
                     $model->{$attribute} = $whatever;
              });
```
